src/Datadog.Trace/ExtensionMethods/DictionaryExtensions.cs

```diff
@@ -13,5 +15,33 @@ internal static class DictionaryExtensions
 
             return default(TValue);
         }
+
+        public static bool TryGetValueOrDefaultAs<TAs>(this IDictionary dictionary, object key, out TAs valueAs)
```

Comment:
The As is because it's being casted to something else (this is a non-generic IDictionary of objects (i.e. a HashTable)).  If the key doesn't exist, you'll get different exceptions/results depending on the concrete implementation (HashTable I believe is just null returned, whereas more modern implementers throw KeyNotFound and/or NotSupported exceptions).

Reply:
Oh, I forgot to remove this `As` comment after we spoke about it over the phone. Feel free to ignore it.

> If the key doesn't exist, you'll get different exceptions/results depending on the concrete implementation

Not if you're using the older non-generic `IDictionary`. [`IDictionary` always returns null when a key is not found](https://docs.microsoft.com/en-us/dotnet/api/system.collections.idictionary.item). In .NET 1.x where there was no way to distinguish between a missing key and an existing key with a null value (which is valid). In .NET 2.0, they decided to fix this by having `IDictionary<TKey, TValue>` throw an exception unless you use `TryGetValue()`, but changing the old `IDictionary` would be a breaking change so we're stuck with that behavior forever.

For example, `Dictionary<TKey, TValue>` (which implements both interfaces) throws an exception when used directly or via the generic interface, but it will return null when used via the non-generic interface. This is intentional to avoid breaking the `IDictionary` contract.

```csharp
var foo = new Dictionary<string,string>();
var value1 = foo ["key"]; // throws exception

IDictionary bar = foo;
var value2 = bar["key"]; // same reference, but returns null
```